subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/internal/DefaultJvmTestSuite.java

```diff
@@ -258,7 +273,24 @@ public void useJUnitJupiter(Provider<String> version) {
 
     @Override
     public void useSpock() {
-        useSpock(Frameworks.SPOCK.defaultVersion);
+        useSpockForDerivedGroovyVersion();
+    }
+
+    private void useSpockForDerivedGroovyVersion() {
+        final Set<String> groovyGroups = new HashSet<>(Arrays.asList("org.codehaus.groovy", "org.apache.groovy"));
+        final Set<String> groovyNames = new HashSet<>(Arrays.asList("groovy", "groovy-all"));
+        Provider<VersionNumber> groovyVersionProvider = getProviderFactory().provider(() -> {
+            Optional<VersionNumber> maybeGroovyJarVersionNumber = runtimeClasspath.getIncoming().getResolutionResult().getAllComponents().stream()
+                .map(ResolvedComponentResult::getModuleVersion)
+                .filter(mvi -> groovyGroups.contains(mvi.getGroup()))
+                .filter(mvi -> groovyNames.contains(mvi.getName()))
+                .map(ModuleVersionIdentifier::getVersion)
+                .map(VersionNumber::parse)
+                .findFirst();
+
+            return maybeGroovyJarVersionNumber.orElse(VersionNumber.parse(GroovySystem.getVersion()));
```

Comment:
Instead of `GroovySystem.getVersion()` should we use `Frameworks.SPOCK.defaultVersion`? Right now the Spock default version is unused. I would prefer to stick with the pattern of using the default version specified by `Frameworks`, even if we choose to set that default version to `GroovySystem.getVersion()`